### PR TITLE
more readme updates

### PR DESCRIPTION
**Pull request by davidosomething**
- Explain purpose of plugin
- Link to minimum required NVIM version commit
- New header for requirements
- Mark `makeprg` as code
- Link to Neovim and syntastic wherever mentioned
- Note that the example maker (jshint) is already included with Neomake
- Replace references to `language` with `filetype` since that is what it works on
- Link out to more of the maker homepages

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,26 +3,29 @@
...
+older versions will probably work fine as well).
 
 ## How to use (basic)
 
 Just set your `makeprg` and `errorformat` as normal, and run:
 
     :Neomake!
 
-If your makeprg can take a filename as an input, then you can run `:Neomake`
+If your `makeprg` can take a filename as an input, then you can run `:Neomake`
```
Comment (blueyed) [inline README.md]: Here it should say `:Neomake!` (with bang), no?

File: README.md [hunk 2]
```diff
@@ -227,15 +230,17 @@ Ruby:
...
 - reek
 - rubylint
 
 Rust:
 
 - rustc
 - cargo (call with `Neomake! cargo`)
-- [clippy](https://github.com/Manishearth/rust-clippy) (call with `Neomake! clippy`, needs nightly rust, supports [rustup](https://github.com/rust-lang-nursery/rustup.rs))
+- [clippy](https://github.com/Manishearth/rust-clippy) (call with
+  `Neomake! clippy`, needs nightly rust, supports
```
Comment (blueyed) [inline README.md]: s/rust/Rust/

File: README.md [hunk 3]
```diff
@@ -227,15 +230,17 @@ Ruby:
...
+- [rubocop](http://batsov.com/rubocop/)
 - reek
 - rubylint
 
 Rust:
 
 - rustc
 - cargo (call with `Neomake! cargo`)
-- [clippy](https://github.com/Manishearth/rust-clippy) (call with `Neomake! clippy`, needs nightly rust, supports [rustup](https://github.com/rust-lang-nursery/rustup.rs))
+- [clippy](https://github.com/Manishearth/rust-clippy) (call with
```
Comment (blueyed) [inline README.md]: s/call with/call it with/

File: README.md [hunk 4]
```diff
@@ -227,15 +230,17 @@ Ruby:
...
+  `Neomake! clippy`, needs nightly rust, supports
+  [rustup](https://github.com/rust-lang-nursery/rustup.rs))
```
Comment (blueyed) [inline README.md]: Dot at the end.

File: README.md (hunk 1 above)
Comment (davidosomething) [inline README.md]: not that one, no (see the text following it)

**Comment by davidosomething** [comment]
- switched "call with" to "run using command" since it is not a function to be called
- link to cargo compiler github page (more informative than the crates.io homepage)

File: README.md (hunk 1 above)
Comment (blueyed) [inline README.md]: I'm confused `:Neomake foo` will not call `make foo`, but gives you: "Neomake: Maker not found: foo".
(`&makeprg` is `make`).

Comment (blueyed) [inline README.md]: This is also what https://github.com/neomake/neomake/issues/589 is about, isn't it?

Comment (davidosomething) [inline README.md]: The documentation for `:h :Neomake` says
```
:Neomake [makers]       Run a make command with the current file as input. If
                        no makers are specified, the default makers for the
                        current |filetype| are used. See |neomake-configuration|
                        for more on makers.
```

The first argument you pass the _command_ is the maker to run.
The maker's `maker.exe` is passed the filename as an arg along with the rest of the args in `maker.args`
So the docs aren't wrong, but this sentence fails to serve its purpose (distinguishing the purpose of `:Neomake` from `:Neomake!`)

Perhaps I'll rewrite this one entirely -- "If your `makeprg` should run on the file in the current buffer, then you can use `:Neomake`" (as opposed to running `:Neomake!` to execute the global maker in the CWD)

**Comment by davidosomething** [comment]
I rewrote the whole usage section

@saaguero -- perhaps move it to the wiki as Syntastic has done?

**Comment by blueyed** [comment]
No time to review it properly now, but how does it compare to https://github.com/neomake/neomake/blob/master/doc/neomake.txt?
I'd like for those to not diverge much, and consider the Vim help to be leading with regard to details and examples.

**Comment by davidosomething** [comment]
no conflicts; the details / examples of recommended use are missing from the `:help` docs, and I would be okay moving them there entirely

**Comment by saaguero** [comment]
@davidosomething 
> perhaps move it to the wiki as Syntastic has done?

Yes, I would rather prefer a collaborative wiki for Makers than having a README.md with these details, what do you think?

**Comment by davidosomething** [comment]
- updated to link to wiki, removed list of included makers

- created wiki page already: https://github.com/neomake/neomake/wiki/Makers

**Comment by saaguero** [comment]
- While on this topic, it would be nice to have a wiki for the README parts about how to add a Maker. Similar to [Syntastic's guide](https://github.com/scrooloose/syntastic/wiki/Syntax-Checker-Guide)

That would be README's sections "Configuration" and "Directory makers".

- Also, it would be nice for the following setting to be under a section called "Recommended settings"
```vim
autocmd! BufWritePost * Neomake
```

I'm towards this setting to be automatically enabled, but that's another discussion, and might be releated to https://github.com/neomake/neomake/issues/405

**Comment by davidosomething** [comment]
Started up wiki docs now, I think this PR is good as is and doesn't contain too much information for an introductory README.md.

So going forward any new information will be separated thusly:

- strictly keep syntax and technical information/configuration options in the `:help` docs
- add all new setup examples, linter info, and anecdotal information in the wiki

File: README.md [hunk 5]
```diff
@@ -76,250 +80,50 @@ Once you have created your makers, run `:Neomake` as normal. Run
...
-
-Vimscript:
-
-- [vint](https://github.com/Kuniwak/vint)
-- [vimlint](https://github.com/syngan/vim-vimlint)
+    :Neomake!
 
-  It can be installed using npm:
-  [node-vimlint](https://www.npmjs.com/package/vimlint).
+In constrast to `:Neomake` without the exclamation, this will execute the
```
Comment (blueyed) [inline README.md]: s/exclamation/exclamation mark/ ?!

File: README.md [hunk 6]
```diff
@@ -76,250 +80,50 @@ Once you have created your makers, run `:Neomake` as normal. Run
...
-- [jsxhint](https://github.com/STRML/JSXHint)
-
-Lua:
-
-- luac
-- luacheck
+Refer to the inline documentation `:h neomake.txt` for more.
 
-Markdown:
+### Directory makers
```
Comment (blueyed) [inline README.md]: They are called project makers also, right?

And we include some already, which could be mentioned.

File: README.md (hunk 1 above)
Comment (blueyed) [inline README.md]: btw: with `NeomakeSh` I was thinking to force the errorformat to accept everything `%+G`, so you will get all the output.

As for #589 defining your own maker for those one-off cases is cumbersome.. that's why I think a new command and or support for e.g. `Neomake! -- clean` (calls `&makeprg foo`) or `Neomake linter --ignore=foo` (calls the linter maker with `--ignore=foo` added to args) should be added.

Not related to this PR, let's discuss it in #589.

**Comment by saaguero** [comment]
I like these changes so far, please take into account blueyed's comments and I think this should be ready to be merged.

On the other hand, it would be really good to have a short gif showing off Neomake, I think just saving some long file and showing that the whole thing runs in async mode would be enough to catch eyes. But this can be track in another PR.

**Comment by davidosomething** [comment]
- Clarify "exclamation" -> "exclamation mark"
- Clarify that some people may say "Project maker"
- Provide examples of Directory makers included with Neomake

File: README.md [hunk 7]
```diff
@@ -102,6 +106,15 @@ defined as a directory maker, e.g., for the `makeclean` example:
 let g:neomake_makeclean_maker = { 'exe': 'make', 'args': ['clean'] }
 ```
 
+An example of a directory makers is the [cargo] for Rust, which you should run
```
Comment (blueyed) [inline README.md]: s/makers/maker/

File: README.md [hunk 8]
```diff
@@ -54,15 +58,15 @@ let g:neomake_javascript_enabled_makers = ['jshint']
...
 
-Where `{ language }` is replaced with the name of the language, and
+Where `{ filetype }` is replaced with the name of the filetype, and
 `{ makername }` is replaced with the name that you want your maker to have. If
 your maker does not follow this convention, Neomake will not be able to see
 it, and you will get an error message like `{ makername } not found`.
 
 Explanation for the strings making up the errorformat can be found by typing
-`:h errorformat` in Neovim (or Vim).
+`:h errorformat` in [Neovim] (or Vim).
```
Comment (saaguero) [inline README.md]: This is rendering an invalid link, I would say to remove the last part, keeping it as:

Explanation for the strings making up the errorformat can be found in `:h erroformat`.

**Comment by davidosomething** [comment]
- fixed invalid link (just refer to Vim)
- fixed plural -> singular

good to go

**Comment by blueyed** [comment]
Huge work, thanks a lot!

**Pull request merged**